p2p/peerstore/testing.go

```diff
@@ -0,0 +1,40 @@
+package peerstore
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+
+	"github.com/libp2p/go-libp2p-core/crypto"
+	"github.com/libp2p/go-libp2p-core/peer"
+
+	ma "github.com/multiformats/go-multiaddr"
+)
+
+func GenerateRandomPeerlist(length int) ([]peer.AddrInfo, error) {
+	peerlist := make([]peer.AddrInfo, 0)
+	for i := 0; i < length; i++ {
+		key, err := rsa.GenerateKey(rand.Reader, 2096)
+		if err != nil {
+			return nil, err
+		}
+
+		_, pubkey, err := crypto.KeyPairFromStdKey(key)
+		if err != nil {
+			return nil, err
+		}
+
+		peerID, err := peer.IDFromPublicKey(pubkey)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
is there a libp2p test utility to do this for you?

Reply:
I couldn't find one, I had to resort to this because simply using random peer IDs breaks JSON unmarshalling for peer.ID type.